### PR TITLE
 Test failure on mining-test.js #3 

### LINKED ISSUE #3: Test failure on mining-test.js
under rewardMiner there are two unit tests fail:
 rewardMiner
    ✔ rewardMiner with 0.5 dcf/s (116ms)
    ✔ rewardMiner with 0.5 dcf/s with put (282ms)
    1) rewardMiner with 0.5 dcf/s and dynamic shards
    2) rewardMiner with 0.5 dcf/y

 1) rewardMiner
       rewardMiner with 0.5 dcf/s and dynamic shards:
     Error: invalid private key (argument="privateKey", value="[[ REDACTED ]]", code=INVALID_ARGUMENT, version=signing-key/5.7.0)
      at Logger.makeError (node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
      at Logger.throwError (node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
      at Logger.throwArgumentError (node_modules/@ethersproject/logger/src.ts/index.ts:285:21)
      at new SigningKey (node_modules/@ethersproject/signing-key/src.ts/index.ts:37:20)
      at new Wallet (node_modules/@ethersproject/wallet/src.ts/index.ts:87:36)
      at Context.<anonymous> (test/mining-test.js:259:18)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

  2) rewardMiner
       rewardMiner with 0.5 dcf/y:
     Error: invalid private key (argument="privateKey", value="[[ REDACTED ]]", code=INVALID_ARGUMENT, version=signing-key/5.7.0)
      at Logger.makeError (node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
      at Logger.throwError (node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
      at Logger.throwArgumentError (node_modules/@ethersproject/logger/src.ts/index.ts:285:21)
      at new SigningKey (node_modules/@ethersproject/signing-key/src.ts/index.ts:37:20)
      at new Wallet (node_modules/@ethersproject/wallet/src.ts/index.ts:87:36)
      at Context.<anonymous> (test/mining-test.js:292:18)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

### PR DESCRIPTION
**Pull request by LiuJiazheng**

### CODE REVIEW

File: contracts/MerkleLib.sol [hunk 1]
```diff
@@ -64,6 +66,7 @@ library MerkleLib {
         uint256 chunkIdx
     ) internal pure returns (bytes32[] memory) {
         uint256 nChunks = 1 << nChunkBits;
+        require(chunkIdx < nChunks, "index must be within the scope of number of chunks");
```
Comment (qizhou) [inline contracts/MerkleLib.sol]: We should reduce the length of the error message to save gas cost.

File: contracts/MerkleLib.sol [hunk 2]
```diff
@@ -46,6 +46,8 @@ library MerkleLib {
         bytes32[] memory proofs
     ) internal pure returns (bool) {
         bytes32 hash = dataHash;
+        if (proofs.length != nChunkBits) return false;
```
Comment (qizhou) [inline contracts/MerkleLib.sol]: Looks like we could remove `nChunkBits` argument and replace it with `proofs.length`?

Comment (LiuJiazheng) [reply contracts/MerkleLib.sol]: Yes, less calldata

File: contracts/MerkleLib.sol (hunk 1 above)
Comment (LiuJiazheng) [reply contracts/MerkleLib.sol]: ACK. will shorten it.

**Review by qizhou — APPROVED** [review]

**Pull request merged**